src/providers/fault-store/fault-store.ts

```diff
@@ -66,4 +77,96 @@ export class FaultStoreProvider {
   getTotals() {
     return this.store.getState();
   }
+
+  /**
+   * This method returns the string 'Pass' or 'Fail' depending on the outcome of
+   * the test.
+   *
+   * It also computes the the numbers of the different types of faults and
+   * stores the representation of the faults required by the test-summary
+   * screen.
+   *
+   * {@link TestSummaryComponent}
+   */
+  getTestResultAndCalculateFaultTotals(): string {
+    let fs = 0;
+    let ds = 0;
+    let ss = 0;
+    this.seriousFaults = [];
+    this.dangerousFaults = [];
+    this.drivingFaults = [];
+    const sectionRegEx = new RegExp(/^[a-z]*/);
+    const data = this.getTotals();
+    Object.keys(data.faults).forEach((fault) => {
+      if (fault === 'lastFault') {
+        return false;
+      }
+      if (data.faults[fault].fault) {
+        fs += data.faults[fault].fault;
+        const section = upperFirst(fault.match(sectionRegEx)[0]);
+        this.drivingFaults.push({
+          name: section + ' - ' + data.faults[fault].faultText,
+          total: data.faults[fault].fault
+        });
+      }
+      if (data.faults[fault].dangerous) {
+        ds += data.faults[fault].dangerous;
+        const section = upperFirst(fault.match(sectionRegEx)[0]);
+        this.dangerousFaults.push({
+          name: section + ' - ' + data.faults[fault].faultText,
+          total: data.faults[fault].fault
+        });
+      }
+      if (data.faults[fault].serious) {
+        ss += data.faults[fault].serious;
+        const section = upperFirst(fault.match(sectionRegEx)[0]);
+        this.seriousFaults.push({
+          name: section + ' - ' + data.faults[fault].faultText,
+          total: data.faults[fault].fault
+        });
+      }
+    });
+
+    this.drivingFaultsNumber = fs;
+    this.seriousFaultsNumber = ss;
+    this.dangerousFaultsNumber = ds;
+
+    if (fs >= 16) {
```

Comment:
can we make one if statement for all fails cases like
```js
if(fs >= 16 || ds > 0 || ss > 0) {
    return TestResult.Fail;
}
```

Reply:
Sure, will do